lib/gui/components/safe-webview/safe-webview.js

```diff
@@ -54,63 +59,76 @@ class SafeWebview extends react.PureComponent {
-
-      return react.createElement('webview', {
-        ref: 'webview',
-        src: url.href
-      }, []);
-    }
-
-    // We have to return null explicitly, undefined is an error in React.
-    return null;
+    return react.createElement('webview', {
+      ref: 'webview',
+      style: {
+        flex: this.state.shouldShow ? null : '0 1',
+        width: this.state.shouldShow ? null : '0',
+        height: this.state.shouldShow ? null : '0'
+      }
+    }, []);
   }
 
   /**
-   * @summary Add the Webview events if there is an element
+   * @summary Add the Webview events
    */
   componentDidMount() {
 
-    // There is no element to add events to if 'shouldLoad' is false.
-    if (this.state.shouldLoad) {
+    // Events React is unaware of have to be handled manually
+    this.refs.webview.addEventListener('did-fail-load', this.didFailLoad);
+    this.refs.webview.addEventListener('did-get-response-details', this.didGetResponseDetails);
+    this.refs.webview.addEventListener('new-window', this.constructor.newWindow);
+    this.refs.webview.addEventListener('console-message', this.constructor.consoleMessage);
 
-      // Events React is unaware of have to be handled manually
-      this.refs.webview.addEventListener('did-fail-load', this.didFailLoad);
-      this.refs.webview.addEventListener('did-get-response-details', this.didGetResponseDetails);
-      this.refs.webview.addEventListener('new-window', this.constructor.newWindow);
-      this.refs.webview.addEventListener('console-message', this.constructor.consoleMessage);
+    // Use the 'success-banner' session
+    this.refs.webview.partition = 'persist:success-banner';
 
-    }
+    // It's important that this comes after the partition setting, otherwise it will
+    // use another session and we can't change it without destroying the element again
+    this.refs.webview.src = this.makeURL();
   }
 
   /**
-   * @summary Remove the Webview events if there is an element
+   * @summary Refresh the webview if we are navigating away from the success page
+   * @param {Object} nextProps - upcoming properties
    */
-  componentWillUnmount() {
+  componentWillReceiveProps(nextProps) {
+    if (this.props.currentStateName === 'success' && nextProps.currentStateName !== 'success') {
 
-    // There is no element to remove events from if 'shouldLoad' is false.
-    if (this.state.shouldLoad) {
+      // Reset URL to initial
+      this.refs.webview.src = this.makeURL();
 
-      // Events React is unaware of have to be handled manually
-      this.refs.webview.removeEventListener('did-fail-load', this.didFailLoad);
-      this.refs.webview.removeEventListener('did-get-response-details', this.didGetResponseDetails);
-      this.refs.webview.removeEventListener('new-window', this.constructor.newWindow);
-      this.refs.webview.removeEventListener('console-message', this.constructor.consoleMessage);
+      const DECISECOND = 100;
```

Comment:
Can we describe what this variable is for instead? e.g. `WEBVIEW_RELOAD_TIMEOUT_MS` or something like that?